src/GraphQL.Tests/Types/AutoRegisteringInterfaceGraphTypeTests.cs

```diff
@@ -0,0 +1,809 @@
+#nullable enable
+
+using System.Collections;
+using System.ComponentModel;
+using System.Reflection;
+using GraphQL.DataLoader;
+using GraphQL.Execution;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+
+namespace GraphQL.Tests.Types;
+
+public class AutoRegisteringInterfaceGraphTypeTests
```

Comment:
Tests are mostly copied from AutoRegisteringObjectGraphTypeTests, modified for interfaces, with a few tests removed and a few added.